extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js

```diff
@@ -125,60 +122,78 @@ export class AmpAdXOriginIframeHandler {
       this.sendEmbedInfo_(this.baseInstance_.isInViewport());
     }));
 
+    // Iframe.onload normally called by the Ad after full load.
+    const iframeLoadPromise = loadPromise(this.iframe).then(() => {
```

Comment:
I don't quite get the change here.
From what I understand, the order should be `load-start` `render-start` `load-end`. 
But here I think `iframeLoadPromise` can be resolved before `renderStartPromise`. Thus it is possible that we resolve `layoutCallback` too soon here?